src/sys/unix.rs

```diff
@@ -1248,21 +1243,26 @@ fn timeval2dur(raw: libc::timeval) -> Option<Duration> {
     }
 }
 
-fn to_s_addr(addr: &Ipv4Addr) -> libc::in_addr_t {
-    let octets = addr.octets();
-    u32::from_ne_bytes(octets)
+pub(crate) fn to_in_addr(addr: &Ipv4Addr) -> in_addr {
+    in_addr {
+        s_addr: u32::from_ne_bytes(addr.octets()),
+    }
 }
 
-fn from_s_addr(in_addr: libc::in_addr_t) -> Ipv4Addr {
-    in_addr.to_be().into()
```

Comment:
`.to_be()` here was not correct really... Well it yielded the same result, but it was not optimally from an understanding point of view IMO. `in_addr` is stored in BE. So `u32::from_be` would be more correct. But both of them just swap the bytes on a LE machine and does nothing on a BE machine, so it does not matter in practice.

Anyway. My solution uses `to_ne_bytes` to avoid a double swap (since the `From<u32>` impl for `Ipv4Addr` swaps the bytes back again).